tests/test_service.py

```diff
@@ -176,7 +139,7 @@ class Bar(metaclass=service.Meta):
 
         with self.assertRaisesRegex(
                 ValueError,
-                "dependency loop: Fnord loops through .*\.(Foo|Bar)"):
+                "dependency loop in service definitions"):
```

Comment:
Ah, pity that we loose the information where the loop goes. That’ll make debugging loops harder. Can we get this back without too much hassle? If it’s too much trouble, don’t bother.